src/main/java/org/powerflows/dmn/engine/evaluator/expression/provider/JuelExpressionEvaluationProvider.java

```diff
@@ -16,27 +16,105 @@
 
 package org.powerflows.dmn.engine.evaluator.expression.provider;
 
+import de.odysseus.el.ExpressionFactoryImpl;
+import lombok.extern.slf4j.Slf4j;
 import org.powerflows.dmn.engine.evaluator.context.EvaluationContext;
+import org.powerflows.dmn.engine.evaluator.expression.provider.binding.ExpressionEvaluationException;
+import org.powerflows.dmn.engine.evaluator.expression.provider.juel.ContextFunctionMapper;
+import org.powerflows.dmn.engine.evaluator.expression.provider.juel.PowerflowsELContext;
 import org.powerflows.dmn.engine.model.decision.field.Input;
 import org.powerflows.dmn.engine.model.decision.rule.entry.InputEntry;
 import org.powerflows.dmn.engine.model.decision.rule.entry.OutputEntry;
 
+import javax.el.ELContext;
+import javax.el.ELException;
+import javax.el.ExpressionFactory;
+import javax.el.FunctionMapper;
 import java.io.Serializable;
 
+@Slf4j
 class JuelExpressionEvaluationProvider implements ExpressionEvaluationProvider {
 
+    private final ExpressionFactory expressionFactory;
+    private final FunctionMapper rootFunctionMapper;
+
+    JuelExpressionEvaluationProvider(final ExpressionEvaluationConfiguration configuration) {
+        this.expressionFactory = new ExpressionFactoryImpl();
+        rootFunctionMapper = new ContextFunctionMapper(configuration.getMethodBinding());
+    }
+
+    private PowerflowsELContext makeContext(final EvaluationContext evaluationContext) {
```

Comment:
More readable is to place public and overriden methods as a first

Reply:
right